### ISSUE TITLE
Manager to honor web map layer's field visibility and order when configured in map viewer FeatureTable

### ISSUE THREAD

**Issue by azizaparveen** [Esri/solutions-components#990]
### Describe requirements

Map viewer now support persisting field visibility and field order set in FeatureTable, we would like Manager to honor these settings when they are configured in map viewer
Here is a test map that has field visibility and order set up on Corvallis HistoricSite layer in Map viewer
https://solutions.mapsdevext.arcgis.com/apps/mapviewer/index.html?webmap=dcbef592ac884dbca6c9516359760950.
Manager currently do not honor these settings.

**Comment by azizaparveen** [comment]
@sarahmcdonald7 do you think we need a config option for this?

**Comment by sarahmcdonald7** [comment]
@azizaparveen I think Manager already honors the configurations set up in Map Viewer. I think we should just show the existing configurations from map viewer in the app configuration. We do this with the filter tool - if layers have filters set up in map viewer already we show them in the expression builder when users go to set up new filters.

**Comment by azizaparveen** [comment]
@sarahmcdonald7 in my testing I don't see Manager is honoring these, Not sure if I missed anything, could please test this as well?
Here is a test app: https://solutions.mapsdevext.arcgis.com/apps/instant/manager/index.html?appid=a1188165b1b24090b9d69191c93466ff
Test map: https://solutions.mapsdevext.arcgis.com/apps/mapviewer/index.html?webmap=dcbef592ac884dbca6c9516359760950.
![image](https://github.com/user-attachments/assets/c918f6dd-0489-4d92-a2ff-03293f69d0b3)
![image](https://github.com/user-attachments/assets/c3680d52-c819-43e7-ad90-30a2fb398e7d)

**Comment by sarahmcdonald7** [comment]
@azizaparveen Odd, I was testing this last week and it was working for me in my app:
<img width="1669" alt="2024-10-29_14-23-40" src="https://github.com/user-attachments/assets/3c28a444-df0c-4721-931a-38571eb6720b">
<img width="1669" alt="2024-10-29_14-23-27" src="https://github.com/user-attachments/assets/c4688e87-15c6-4b13-839e-38b41abdd9cf">
https://holistic.mapsdevext.arcgis.com/apps/instant/manager/index.html?appid=3a19c9f52d2745599fb3c04d281ea1e6

**Comment by sarahmcdonald7** [comment]
@azizaparveen I know right now field descriptions are being displayed in the table. Lucy suggested hiding descriptions behind an "i" icon. 
![2024-12-04_13-59-20](https://github.com/user-attachments/assets/39e5a54e-b416-438e-bced-dad3f82338ef)

**Comment by azizaparveen** [comment]
@sarahmcdonald7 liked the idea Lucy suggested

**Comment by azizaparveen** [comment]
So basically we need to do the following:

- [ ] Manager to honor Web map layer's field visibility when set in the Table
- [ ] Manager to honor Web map player's field order when set in the table
- [ ] Manager to show/hide field description using i icon as suggested above.

**Comment by sumitzarkar** [comment]
@azizaparveen 

>  Manager to honor Web map layer's field visibility when set in the Table
 Manager to honor Web map player's field order when set in the table
- Currently, the manager component shows the fields and the field order using the configuration that we configure in the instant app settings
 ![image](https://github.com/user-attachments/assets/443a4f4c-1d04-4369-8c78-bd9d99353dc1)

- When any field is removed or reordered in the table using Mapviewer this information is stored in **attributeTableInfo** property of the layer in web map details
![image](https://github.com/user-attachments/assets/6117aa37-df93-471e-a43f-527a8350a350)

If we update the code to show the fields according to the map configuration, then the instant app settings and the component runtime will not be in sync.

> Manager to show/hide field description using i icon as suggested above.
- It would be better if we show the info icon before the column header title as it will look more aligned in all columns
   ![image](https://github.com/user-attachments/assets/650a6c28-ae69-4d59-9d1a-0a17ff8a2d20)

- We can add an icon in the table header. Now we are looking to show the field description on hover of the icon.

**Comment by sumitzarkar** [comment]
@azizaparveen 

As discussed on the call and according tothe  above comment:

>Manager to honor Web map layer's field visibility when set in the Table
Manager to honor Web map player's field order when set in the table

- Manager component will always show fields and their order based on the configuration from the instant app
- Instant app configuration should reflect the webmaps fields and order using the attributeTableInfo. 
- So this needs  changes only in the instant app configuration

>Manager to show/hide field description using i icon as suggested above.
- We can show the informationIcon using the icon property of the [FieldColumnTemplate ](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-FeatureTable-support-FieldColumnTemplate.html)
- We need the JS API team's input to understand how to show the field description on the hover of this icon. Or if this functionality can be easily implemented in JS API itself

**Comment by azizaparveen** [comment]
Submitted an issue to take care of the field order and visibility in the configuration side https://devtopia.esri.com/WebGIS/arcgis-portal-app-templates/issues/11257

**Comment by azizaparveen** [comment]
> > Manager to show/hide field description using i icon as suggested above.
> 
> * We can show the informationIcon using the icon property of the [FieldColumnTemplate ](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-FeatureTable-support-FieldColumnTemplate.html)
> * We need the JS API team's input to understand how to show the field description on the hover of this icon. Or if this functionality can be easily implemented in JS API itself

Sent an email to FeatureTable tem to get more info on this.

**Comment by azizaparveen** [comment]
@sumitzarkar John is going to work on this issue on the configuration side.

**Comment by azizaparveen** [comment]
@sumitzarkar heard back from JS API team on showing field description on the hover of info icon. This is not currently supported but they are going to add a new property. Here is the detail, I will check back with him again next week on the progress.

"This is not currently supported OOTB on production but I will add a new property to the column template definition to allow passing in text to be assigned to the icon. If no text is provided, we will default to showing the field alias/name on hover as it does now. I will install it this week and it will be available in the 4.32 release/jsdev branch.

Example:

const table = new FeatureTable({
  // ...
  tableTemplate: {
    columnTemplates: [
        {
          type: "field",
          fieldName: "NAME",
          label: "School Name",
          icon: "education",
          iconText: "Custom text for icon",
        },
        { 
          // ... field2
        }
    ]
  }
});

**Comment by azizaparveen** [comment]
Logged a new issue for the field description enhancement. https://github.com/Esri/solutions-components/issues/1090

**Comment by azizaparveen** [comment]
@jmhauck are you talking about this two options? it makes sense to me to extend Manager configuration to support these, do you have any idea what will be the level of effort needed for this? Also, if we used Field selector component in Manager, in that case, the component needs to be updated right? 

![Image](https://github.com/user-attachments/assets/e2227c6e-08a0-4791-8677-2778447cc8a4)

**Comment by jmhauck** [comment]
@azizaparveen Yes these are the options and yes the solutions component would need to be updated also to support these at runtime as well. I'm not really sure on LOE yet...still looking at it but wanted to bring this up to you all.

**Comment by azizaparveen** [comment]
> [@azizaparveen](https://github.com/azizaparveen) Yes these are the options and yes the solutions component would need to be updated also to support these at runtime as well. I'm not really sure on LOE yet...still looking at it but wanted to bring this up to you all.

Thanks for looking into it @jmhauck, if this gets to big to handle in R1, we can push it to R2

**Comment by sumitzarkar** [comment]
@azizaparveen 

I guess we need changes for this only in the instant app configuration, no changes required in the manager component.

@jmhauck  - please confirm if we are missing any thing in this and if you see any changes to be done from our side

**Comment by azizaparveen** [comment]
@sumitzarkar hopefully this will be ready by the end of the week.

**Comment by azizaparveen** [comment]
@sumitzarkar the configuration is available in devext 

![Image](https://github.com/user-attachments/assets/0ee384fc-1dde-411e-8280-0d454ec4511d)

**Comment by sumitzarkar** [comment]
@jmhauck @azizaparveen 

We see this configuration in DevEXT. However, we are unclear how the layer visibility and the fields sequence and visibility will be passed to the Manager component.

Currently, the Manager shows all the Layers/Tables and fields from mapInfo

![Image](https://github.com/user-attachments/assets/83f5e730-d568-4089-8675-baa28dcbafdb)

@jmhauck Could you guide us if any changes are required from CT on this or if this needs to be updated in the InstantAPP to pass the new configuration in the existing manager properties
